src/gui/sdl_mapper.cpp

```diff
@@ -1873,9 +1910,11 @@ class CHandlerEvent final : public CTriggeredEvent {
 
 	void Active(bool yesno) override { (*handler)(yesno); }
 
-	bool HasSameBinding(const CKeyBind* other) const
+	bool HasSameBinding(const CBind* other) const
 	{
-		return other && defkey == other->key && defmod == other->mods;
+		auto other_keybind = dynamic_cast<const CKeyBind*>(other);
+		return other_keybind && defkey == other_keybind->key &&
+		       defmod == other->mods;
```

Comment:
Except here you don't have the separate if statement.  Definitely prefer consistency one way or the other.